src/parser/Iri.h

```diff
@@ -48,6 +48,13 @@ class Iri {
   // Return the string value of the iri object without any leading or trailing
   // angled brackets.
   NormalizedStringView getContent() const;
+
+  // Calculate the memory usage of the `Iri` string. This might overestimate the
+  // memory usage as this does not currently take into account small string
+  // optimization of `std::string`
+  size_t getDynamicMemoryUsage() const {
+    return sizeof(std::string) + iri_.capacity();
```

Comment:
We have to make clear, whether the hash set expects only the dynamic part of the memory usage, without the `sizeof()` part (probably yes, as this is managed differently by a lot of containers).

Reply:
Removed this from `Iri.h` and `Literal.h`. I think the allocator (I am not sure if I could use the exisiting `AllocatorWithLimit` or if I need to write a new one??) could keep track of the static memory while the `SizeGetter`s just return the dynamic memory usage.